resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala

```diff
@@ -149,7 +151,7 @@ private[yarn] class YarnAllocator(
   // Resource capability requested for each executor
   private[yarn] val resource: Resource = {
     val resource = Resource.newInstance(
-      executorMemory + memoryOverhead + pysparkWorkerMemory, executorCores)
+      executorMemory + executorOffHeapMemory + memoryOverhead + pysparkWorkerMemory, executorCores)
```

Comment:
According line 258 to 260, `memoryOverhead` includes `pysparkWorkerMemory`, but looks difference here.

Reply:
Yes, this makes me confused.

Reply:
Isn't this what `spark.executor.memoryOverhead` already adds via `memoryOverhead`?
I guess I'm wondering what `MEMORY_OFFHEAP_SIZE` does that's supposed to be different.

Reply:
@srowen `memoryOverhead` include `MEMORY_OFFHEAP_SIZE `  before this pr, and `memoryOverhead` and `MEMORY_OFFHEAP_SIZE ` had to be modified at the same time to ensure request enough resources from yarn if I want to increase `MEMORY_OFFHEAP_SIZE `, this is not user friendly. This pr let them be independent.